compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7024,6 +7029,28 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
     }
 
+    private void populateIncludedRecordParams(BVarSymbol param, List<BRecordType> incRecords) {
+        if (Symbols.isFlagOn(param.flags, Flags.INCLUDED)) {
+            incRecords.add((BRecordType) Types.getReferredType(param.type));
+        }
+    }
+
+    // If there is a named-arg or positional-arg corresponding to an included-record-param,
+    // it is an error for a named-arg to specify a field of that included-record-param.
+    private void checkSameNamedArgsInIncRecords(List<BLangExpression> namedArgs, List<BRecordType> incRecords) {
+        for (BLangExpression namedArg : namedArgs) {
+            String argName = ((NamedArgNode) namedArg).getName().value;
+            for (BRecordType record : incRecords) {
+                if (record.fields.containsKey(argName)) {
```

Comment:
This check isn't correct. It disallows the following also, but it should work.

```ballerina
type Rec record {|
    never a?;
    int b;
|};

function fn(string a, *Rec rec) {

}

public function main() {
    fn(a = "", rec = {b: 2});
}
```

Reply:
Please add a test also.

Reply:
Added